vortex-file/src/read/layouts/chunked.rs

```diff
@@ -111,11 +140,15 @@ impl ChunkedLayout {
         self.children()
             .zip_eq(self.child_ranges())
             .map(|((i, c), (begin, end))| {
+                let layout_part_id = i as u16;
+                if layout_part_id == METADATA_LAYOUT_PART_ID {
+                    vortex_bail!("metadata should not be among children");
+                }
```

Comment:
it seems weird to throw here if we filter it out in `self.children()`

Reply:
Removed.